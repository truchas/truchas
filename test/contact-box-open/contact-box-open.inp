&OUTPUTS
  output_t = 0.0, 100.0, 150.0
  output_dt = 5.0, 50.0
  output_dt_multiplier = 1
  !short_output_dt_multiplier = 1
  !long_output_dt_multiplier = 5
  !Long_Edit_Bounding_Coords = 0.0, 8.0e-2, 0.00, 8.0e-2, 0.00, 8.0e-2
/

&MESH
  mesh_file = 'split-rotated-cube.exo'
  coordinate_scale_factor = 0.01
  gap_element_blocks = 3
/

&PHYSICS
  fluid_flow = .false.
  heat_transport = .true.
  solid_mechanics = .true.
/

&DIFFUSION_SOLVER
  abs_temp_tol = 0.0
  rel_temp_tol = 5.0e-5
  abs_enthalpy_tol = 0.0
  rel_enthalpy_tol = 5.0e-5
  nlk_tol = 0.05
  nlk_preconditioner = 'hypre_amg'
  pc_amg_cycles = 2
  verbose_stepping = .true.
/

&NUMERICS
  dt_init = 1.0
  dt_grow = 5.0
  dt_max = 5.0
  dt_min = 1.0e-3
  displacement_nonlinear_solution = 'displacement solve'
  contact_distance = 1.0e-7
  contact_norm_trac = 1.0e4
  contact_penalty = 1.0e3
/

&LINEAR_SOLVER
  name = 'displacement precon'
  method = 'none'
  preconditioning_steps = 4
  relaxation_parameter = 1.6
  preconditioning_method = 'tm_ssor'
/

&NONLINEAR_SOLVER 
  name = 'displacement solve'
  method = 'nlk'
  linear_solver_name = 'displacement precon'
  convergence_criterion = 1.0e-12
  NLK_max_vectors = 10
  NLK_vector_tolerance = 0.001
  maximum_iterations = 2000
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 1
  material_number = 1
  temperature = 308.0
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 2
  material_number = 1
  temperature = 308.0
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 3
  material_number = 2
  temperature = 308.0
/

&DS_BOUNDARY_CONDITION
  name = 'ends'
  variable = 'temperature'
  condition = 'dirichlet'
  face_set_ids = 1, 2
  data_function = 'applied temperature profile'
/

&FUNCTION
  name = 'applied temperature profile'
  type = 'smooth step'
  smooth_step_x0 = 0.0
  smooth_step_y0 = 308.0
  smooth_step_x1 = 100.0
  smooth_step_y1 = 288.0
/

&DS_BOUNDARY_CONDITION
  name = 'lateral sides'
  variable = 'temperature'
  condition = 'flux'
  face_set_ids = 3, 4, 5, 6
  data_constant = 0.0
/

&DS_INTERFACE_CONDITION
  name = 'gap interface'
  variable = 'temperature'
  condition = 'HTC'
  face_set_ids = 3
  data_constant = 1.0e4
/

&BC
  bc_name = 'left end -- no normal displacement'
  surface_name = 'from mesh file'
  mesh_surface = 1
  bc_variable = 'displacement'
  bc_type = 'normal-displacement' 
  bc_value = 0.0
/

&BC
  bc_name = 'gap interface'
  surface_name = 'from mesh file'
  mesh_surface = 10
  bc_variable = 'displacement'
  bc_type = 'contact'
  bc_value = 0.0
/

&BC
  bc_name = 'right end -- no normal displacement'
  surface_name = 'from mesh file'
  mesh_surface = 2
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

&BC
  bc_name = 'front side'
  surface_name = 'from mesh file'
  mesh_surface = 3
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

&BC
  bc_name = 'bottom side'
  surface_name = 'from mesh file'
  mesh_surface = 5
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

&BC
  bc_name = 'top side'
  surface_name = 'from mesh file'
  mesh_surface = 6
  bc_variable = 'displacement'
  bc_type = 'normal-displacement'
  bc_value = 0.0
/

define the material properties (aluminum, SI units)
Elastic constants (stresses) in MPa to make test checking easier 
&MATERIAL
  material_number =  1
  priority =  1
  material_name = 'aluminum'
  material_feature = 'background'
  density =  2.70e+03
  immobile = .true.
/
  Lame1_Constants =  5.20e+10
  Lame2_Constants =  2.60e+10

&PHASE
  name = 'aluminum'
  property_name(1) = 'density',       property_constant(1) = 2.7e3
  property_name(2) = 'specific heat', property_constant(2) = 9.0e2
  property_name(3) = 'conductivity',  property_constant(3) = 2.4e2
  property_name(4) = 'TM reference density',     property_constant(4) = 2.7e3
  property_name(5) = 'TM reference temperature', property_constant(5) = 298.0
  property_name(6) = 'TM linear CTE',            property_constant(6) = 2.2e-5
  property_name(7) = 'Lame1', property_constant(7) = 5.20e+10
  property_name(8) = 'Lame2', property_constant(8) = 2.60e+10
/

&MATERIAL_SYSTEM
  name = 'aluminum'
  phases = 'aluminum'
/

&MATERIAL
  material_number = 2
  priority = 2
  material_name = 'gap'
  density =  0.0
  immobile = .true.
/
  Lame1_Constants = 0.0
  Lame2_Constants = 0.0
