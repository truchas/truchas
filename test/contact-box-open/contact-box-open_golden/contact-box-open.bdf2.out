
TRAP step      1: T= 0.00000E+00, H= 1.00000E+00, ETAH= 5.00000E-01
  Preconditioner updated at T= 1.00000E+00, ETAH= 5.00000E-01
    1: error= 1.31190E-01
    2: error= 4.90811E-03
  AIN BCE solve succeeded:   2 iterations, error= 4.90811E-03
  Step accepted: no local error control

BDF2 step      2: T= 1.00000E+00, H= 1.00000E+00, ETAH= 6.66667E-01
    1: error= 7.63695E-01
    2: error= 3.62040E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.62040E-02
  Step accepted: no local error control

BDF2 step      3: T= 2.00000E+00, H= 1.00000E+00, ETAH= 6.66667E-01
    1: error= 1.23441E-01
    2: error= 3.77344E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.77344E-02
  Step accepted: no local error control

BDF2 step      4: T= 3.00000E+00, H= 1.00000E+00, ETAH= 6.66667E-01
    1: error= 9.62306E-02
    2: error= 2.77962E-02
  AIN BCE solve succeeded:   2 iterations, error= 2.77962E-02
  Step accepted: perr= 1.21896E-01

BDF2 step      5: T= 4.00000E+00, H= 2.02331E+00, ETAH= 1.21212E+00
    1: error= 1.65829E-01
    2: error= 1.49509E-01
    3: error= 3.38960E-03
  AIN BCE solve succeeded:   3 iterations, error= 3.38960E-03
  Step accepted: perr= 3.12444E-01

BDF2 step      6: T= 6.02331E+00, H= 1.97274E+00, ETAH= 1.32073E+00
    1: error= 1.02419E-01
    2: error= 6.03076E-02
    3: error= 6.70153E-03
  AIN BCE solve succeeded:   3 iterations, error= 6.70153E-03
  Step accepted: perr= 1.35817E-01

BDF2 step      7: T= 7.99605E+00, H= 3.51624E+00, ETAH= 2.14327E+00
  Preconditioner updated at T= 1.15123E+01, ETAH= 2.14327E+00
    1: error= 3.77502E-01
    2: error= 1.37429E-02
  AIN BCE solve succeeded:   2 iterations, error= 1.37429E-02
  Step accepted: perr= 3.77502E-01

BDF2 step      8: T= 1.15123E+01, H= 3.24880E+00, ETAH= 2.19479E+00
    1: error= 5.00724E-01
    2: error= 3.72582E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.72582E-02
  Step accepted: perr= 5.00724E-01

BDF2 step      9: T= 1.47611E+01, H= 3.07872E+00, ETAH= 2.07104E+00
    1: error= 5.00822E-01
    2: error= 3.35544E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.35544E-02
  Step accepted: perr= 5.00822E-01

BDF2 step     10: T= 1.78398E+01, H= 3.20020E+00, ETAH= 2.11980E+00
    1: error= 5.15128E-01
    2: error= 4.05467E-02
  AIN BCE solve succeeded:   2 iterations, error= 4.05467E-02
  Step accepted: perr= 5.13440E-01

BDF2 step     11: T= 2.10400E+01, H= 3.13012E+00, ETAH= 2.09448E+00
    1: error= 4.95668E-01
    2: error= 3.71646E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.71646E-02
  Step accepted: perr= 4.95138E-01

BDF2 step     12: T= 2.41701E+01, H= 3.15663E+00, ETAH= 2.10146E+00
    1: error= 5.01292E-01
    2: error= 3.80895E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.80895E-02
  Step accepted: perr= 4.99741E-01

BDF2 step     13: T= 2.73268E+01, H= 3.15820E+00, ETAH= 2.10529E+00
    1: error= 5.00554E-01
    2: error= 3.83369E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.83369E-02
  Step accepted: perr= 4.97427E-01

BDF2 step     14: T= 3.04850E+01, H= 3.16155E+00, ETAH= 2.10733E+00
    1: error= 5.03446E-01
    2: error= 3.86839E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.86839E-02
  Step accepted: perr= 5.00351E-01

BDF2 step     15: T= 3.36465E+01, H= 3.15853E+00, ETAH= 2.10602E+00
    1: error= 5.04606E-01
    2: error= 3.86658E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.86658E-02
  Step accepted: perr= 5.01432E-01

BDF2 step     16: T= 3.68051E+01, H= 3.15437E+00, ETAH= 2.10337E+00
    1: error= 5.05005E-01
    2: error= 3.84838E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.84838E-02
  Step accepted: perr= 5.01487E-01

BDF2 step     17: T= 3.99594E+01, H= 3.15170E+00, ETAH= 2.10143E+00
    1: error= 5.05317E-01
    2: error= 3.83539E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.83539E-02
  Step accepted: perr= 5.01530E-01

BDF2 step     18: T= 4.31111E+01, H= 3.14842E+00, ETAH= 2.09931E+00
    1: error= 5.05458E-01
    2: error= 3.81953E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.81953E-02
  Step accepted: perr= 5.01560E-01

BDF2 step     19: T= 4.62595E+01, H= 3.14504E+00, ETAH= 2.09707E+00
    1: error= 5.05639E-01
    2: error= 3.80283E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.80283E-02
  Step accepted: perr= 5.01579E-01

BDF2 step     20: T= 4.94046E+01, H= 3.14177E+00, ETAH= 2.09487E+00
    1: error= 5.05833E-01
    2: error= 3.78658E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.78658E-02
  Step accepted: perr= 5.01584E-01

BDF2 step     21: T= 5.25463E+01, H= 3.13845E+00, ETAH= 2.09267E+00
    1: error= 5.06013E-01
    2: error= 3.77010E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.77010E-02
  Step accepted: perr= 5.01578E-01

BDF2 step     22: T= 5.56848E+01, H= 3.13516E+00, ETAH= 2.09047E+00
    1: error= 5.06189E-01
    2: error= 3.75365E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.75365E-02
  Step accepted: perr= 5.01558E-01

BDF2 step     23: T= 5.88200E+01, H= 3.13194E+00, ETAH= 2.08832E+00
    1: error= 5.06355E-01
    2: error= 3.73742E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.73742E-02
  Step accepted: perr= 5.01526E-01

BDF2 step     24: T= 6.19519E+01, H= 3.12879E+00, ETAH= 2.08621E+00
    1: error= 5.06509E-01
    2: error= 3.72149E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.72149E-02
  Step accepted: perr= 5.01481E-01

BDF2 step     25: T= 6.50807E+01, H= 3.12576E+00, ETAH= 2.08418E+00
    1: error= 5.06650E-01
    2: error= 3.70600E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.70600E-02
  Step accepted: perr= 5.01424E-01

BDF2 step     26: T= 6.82064E+01, H= 3.12286E+00, ETAH= 2.08223E+00
    1: error= 5.06776E-01
    2: error= 3.69108E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.69108E-02
  Step accepted: perr= 5.01353E-01

BDF2 step     27: T= 7.13293E+01, H= 3.12013E+00, ETAH= 2.08039E+00
    1: error= 5.06885E-01
    2: error= 3.67688E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.67688E-02
  Step accepted: perr= 5.01270E-01

BDF2 step     28: T= 7.44494E+01, H= 3.11758E+00, ETAH= 2.07867E+00
    1: error= 5.06976E-01
    2: error= 3.66352E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.66352E-02
  Step accepted: perr= 5.01174E-01

BDF2 step     29: T= 7.75670E+01, H= 3.11525E+00, ETAH= 2.07709E+00
    1: error= 5.07047E-01
    2: error= 3.65115E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.65115E-02
  Step accepted: perr= 5.01065E-01

BDF2 step     30: T= 8.06823E+01, H= 3.11316E+00, ETAH= 2.07567E+00
    1: error= 5.07097E-01
    2: error= 3.63990E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.63990E-02
  Step accepted: perr= 5.00942E-01

BDF2 step     31: T= 8.37954E+01, H= 3.11134E+00, ETAH= 2.07443E+00
    1: error= 5.07124E-01
    2: error= 3.62991E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.62991E-02
  Step accepted: perr= 5.00807E-01

BDF2 step     32: T= 8.69068E+01, H= 3.10981E+00, ETAH= 2.07338E+00
    1: error= 5.07127E-01
    2: error= 3.62132E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.62132E-02
  Step accepted: perr= 5.00660E-01

BDF2 step     33: T= 9.00166E+01, H= 3.10860E+00, ETAH= 2.07253E+00
    1: error= 5.07104E-01
    2: error= 3.61426E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.61426E-02
  Step accepted: perr= 5.00499E-01

BDF2 step     34: T= 9.31252E+01, H= 3.10774E+00, ETAH= 2.07192E+00
    1: error= 5.07053E-01
    2: error= 3.60887E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.60887E-02
  Step accepted: perr= 5.00326E-01

BDF2 step     35: T= 9.62329E+01, H= 3.10724E+00, ETAH= 2.07155E+00
    1: error= 5.06973E-01
    2: error= 3.60528E-02
  AIN BCE solve succeeded:   2 iterations, error= 3.60528E-02
  Step accepted: perr= 5.00140E-01

BDF2 step     36: T= 9.93402E+01, H= 3.10715E+00, ETAH= 2.07144E+00
    1: error= 2.03626E+00
    2: error= 7.01542E-02
    3: error= 6.65803E-03
  AIN BCE solve succeeded:   3 iterations, error= 6.65803E-03
  Step accepted: perr= 2.03626E+00

BDF2 step     37: T= 1.02447E+02, H= 1.32344E+00, ETAH= 1.01905E+00
    1: error= 1.56381E+00
    2: error= 3.90771E-01
    3: error= 2.67798E-02
  AIN BCE solve succeeded:   3 iterations, error= 2.67798E-02
  Step accepted: perr= 1.56381E+00

BDF2 step     38: T= 1.03771E+02, H= 1.07208E+00, ETAH= 7.40623E-01
    1: error= 5.98085E-01
    2: error= 3.11275E-01
    3: error= 1.44220E-02
  AIN BCE solve succeeded:   3 iterations, error= 1.44220E-02
  Step accepted: perr= 3.09173E-01

BDF2 step     39: T= 1.04843E+02, H= 1.84624E+00, ETAH= 1.13083E+00
    1: error= 6.32131E-01
    2: error= 1.82875E-01
    3: error= 1.70385E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.70385E-03
  Step accepted: perr= 4.53660E-01

BDF2 step     40: T= 1.06689E+02, H= 1.60968E+00, ETAH= 1.09817E+00
    1: error= 1.88628E-01
    2: error= 5.34458E-02
    3: error= 1.72816E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.72816E-03
  Step accepted: perr= 1.35579E-01

BDF2 step     41: T= 1.08299E+02, H= 3.05669E+00, ETAH= 1.84689E+00
    1: error= 1.68806E-01
    2: error= 8.22417E-03
  AIN BCE solve succeeded:   2 iterations, error= 8.22417E-03
  Step accepted: perr= 1.60614E-01

BDF2 step     42: T= 1.11355E+02, H= 4.34046E+00, ETAH= 2.73540E+00
    1: error= 2.64994E-02
    2: error= 4.70875E-03
  AIN BCE solve succeeded:   2 iterations, error= 4.70875E-03
  Step accepted: perr= 3.11484E-02

BDF2 step     43: T= 1.15696E+02, H= 5.00000E+00, ETAH= 3.25668E+00
    1: error= 1.78553E-02
    2: error= 4.50623E-03
  AIN BCE solve succeeded:   2 iterations, error= 4.50623E-03
  Step accepted: perr= 2.22760E-02

BDF2 step     44: T= 1.20696E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 1.17623E-02
    2: error= 3.13994E-03
  AIN BCE solve succeeded:   2 iterations, error= 3.13994E-03
  Step accepted: perr= 1.46656E-02

BDF2 step     45: T= 1.25696E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 3.81290E-03
    2: error= 9.99226E-04
  AIN BCE solve succeeded:   2 iterations, error= 9.99226E-04
  Step accepted: perr= 4.71787E-03

BDF2 step     46: T= 1.30696E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 4.44118E-04
  AIN BCE solve succeeded:   1 iterations, error= 4.44118E-04
  Step accepted: perr= 4.44118E-04

BDF2 step     47: T= 1.35696E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 5.03591E-05
  AIN BCE solve succeeded:   1 iterations, error= 5.03591E-05
  Step accepted: perr= 5.03591E-05

BDF2 step     48: T= 1.40696E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 3.42185E-04
  AIN BCE solve succeeded:   1 iterations, error= 3.42185E-04
  Step accepted: perr= 3.42185E-04

BDF2 step     49: T= 1.45696E+02, H= 5.00000E+00, ETAH= 3.33333E+00
    1: error= 2.26401E-04
  AIN BCE solve succeeded:   1 iterations, error= 2.26401E-04
  Step accepted: perr= 2.26401E-04
